### ISSUE TITLE
Occluded hand joints

### ISSUE THREAD

**Issue by anjugopinath** [zc-alexfan/arctic#21]
Is it possible to know which hand joints are occluded in each image in the ground truth training data?

**Comment by zc-alexfan** [comment]
Occluded by the object?

**Comment by anjugopinath** [comment]
Any type of occlusion - self-occlusion, occlusion by an object, or occlusion by the other hand.

**Comment by zc-alexfan** [comment]
Hand/hand occlusion, and hand-object occlusion, you can render the segmentation mask of the data. Then the mask value will be the foreground entity. 

For self-occlusion of hands, it is a bit tricky. It really depends on the granularity of your "occlusion" definition. For example, when a part of the index finger tip is occluded, do you want to assign the finger joint to be occluded? or you want to decide whether each vertex is occluded. 

You can still render masks to decide what is occluded. How fine, the mask is defined, controls how precise occlusion is defined. 

Here are several ways to define the granularity of hand segmentation:
- Joint-level
- Part-level ([see Fig. 4](https://arxiv.org/pdf/2107.00434.pdf))
- Vertex-level (http://densepose.org/)

For part-level, we have implemented code, but it is not implemented in the ARCTIC framework. Let me know if you want to try.

**Comment by anjugopinath** [comment]
Thank You so much for the detailed response.
I want to get an estimate of the % of occlusion in the entire dataset. This would be faster if implemented with code. So, can I implement this with code?

**Comment by zc-alexfan** [comment]
Sorry, I am confused by the question. Can you clarify?

**Comment by heydarshahi** [comment]
Sorry for re-opening this issue; I'm also trying to spot joint occlusions using your earlier response. However, I couldn't fully understand your method:

> You can render the segmentation mask of the data. Then the mask value will be the foreground entity (entities without a foreground value will be occluded).

What do you mean by "foreground entity"? How can I check if the segmentation mask value is the foreground entity? 

I'd highly appreciate your input on this. 

Thanks!
Amin

**Comment by zc-alexfan** [comment]
Here I refer to part segmentation of the hand. You use a textured map to indicate different hand parts. Then render those hand parts. The rendered pixel color will be the part that is at the front. 

See https://github.com/zc-alexfan/digit-interacting paper for hand parts.
